runtime/compiler/env/J9SharedCache.cpp

```diff
@@ -885,20 +885,21 @@ TR_J9SharedCache::rememberClass(J9Class *clazz, const AOTCacheClassChainRecord *
    chainData = findChainForClass(clazz, key, keyLength);
    if (chainData != NULL)
       {
+      uintptr_t chainOffset = TR_SharedCache::INVALID_CLASS_CHAIN_OFFSET;
       if (classMatchesCachedVersion(clazz, chainData))
          {
          LOG(1, "\tcurrent class and class chain found (%p) are identical; returning the class chain\n", chainData);
+         chainOffset = offsetInSharedCacheFromPointer(chainData);
```

Comment:
Please have a look at https://github.com/eclipse-openj9/openj9/pull/15700 and determine whether we should call `isPointerInSharedCache()` instead.

Reply:
We should call it, then.

Reply:
We should call `isPointerInSharedCache()` if we've just stored something in the local SCC and want to find its offset. Here, we've found a pre-existing chain in the local SCC with `findChainForClass`, so I think `offsetInSharedCacheFromPointer()` will always succeed in determining the offset of `chainData`.

Later on in the method, after we've created and stored a new class chain for `clazz`, we definitely need to use `isPointerInSharedCache()`, as that's exactly the situation the linked PR addressed.